src/internal/signalr/handler.go

```diff
@@ -26,36 +27,40 @@ type observationsHandler struct {
 	meterElec   numericmeter.Service
 	cache       cache.Cache
 	callbacks   map[ObservationID]func(Observation) error
+	thing       adapter.Thing
 
 	isCloudOnline atomic.Bool
 	isStateOnline atomic.Bool
 }
 
 // NewObservationsHandler creates new observation handler.
 func NewObservationsHandler(thing adapter.Thing, cache cache.Cache) (Handler, error) {
-	chargepoint, err := getChargepointService(thing)
-	if err != nil {
-		return nil, err
+	handler := observationsHandler{
+		cache: cache,
+		thing: thing,
 	}
 
-	meterElec, err := getMeterElecService(thing)
+	chargepoint, err := handler.getChargepointService()
 	if err != nil {
 		return nil, err
 	}
 
-	handler := observationsHandler{
-		chargepoint: chargepoint,
-		meterElec:   meterElec,
-		cache:       cache,
+	meterElec, err := handler.getMeterElecService()
+	if err != nil {
+		return nil, err
 	}
 
+	handler.chargepoint = chargepoint
```

Comment:
As we discussed earlier - you need to remove references to `chargepoint` and `meterElec` and utilize `get` methods instead.